f5_openstack_agent/lbaasv2/drivers/bigip/tenants.py

```diff
@@ -82,59 +97,44 @@ def assure_tenant_cleanup(self, service, all_subnet_hints):
     # otherwise called once
     def _assure_bigip_tenant_cleanup(self, bigip, service, subnet_hints):
         tenant_id = service['loadbalancer']['tenant_id']
-        if self.conf.f5_sync_mode == 'replication':
-            self._remove_tenant_replication_mode(bigip, tenant_id)
-        else:
-            self._remove_tenant_autosync_mode(bigip, tenant_id)
+
+        self._remove_tenant_replication_mode(bigip, tenant_id)
 
     def _remove_tenant_replication_mode(self, bigip, tenant_id):
         # Remove tenant in replication sync-mode
         partition = self.service_adapter.get_folder_name(tenant_id)
         domain_names = self.network_helper.get_route_domain_names(bigip,
                                                                   partition)
-        if self.driver.disconnected_service.network_exists(bigip, partition):
-            self.driver.disconnected_service.delete_network(bigip, partition)
-
-        if domain_names:
-            for domain_name in domain_names:
+        for domain_name in domain_names:
+            try:
                 self.network_helper.delete_route_domain(bigip,
                                                         partition,
                                                         domain_name)
-        # sudslog = std_logging.getLogger('suds.client')
-        # sudslog.setLevel(std_logging.FATAL)
-        self.system_helper.force_root_folder(bigip)
-        # sudslog.setLevel(std_logging.ERROR)
-
-        try:
-            self.system_helper.delete_folder(bigip, partition)
-        except f5ex.SystemDeleteException:
-            self.system_helper.purge_folder_contents(bigip, partition)
-            self.system_helper.delete_folder(bigip, partition)
-
-    def _remove_tenant_autosync_mode(self, bigip, tenant_id):
-        partition = self.service_adapter.get_folder_name(tenant_id)
+            except Exception as err:
+                LOG.error("Failed to delete route domain %s. "
+                          "%s. Manual intervention might be required."
+                          % (domain_name, err.message))
+            if self.driver.disconnected_service.network_exists(
```

Comment:
The if block starting at line 117 should not be nested inside of the for loop.  There is only one disconnected network per tenant.  It used to be located at the top level before purging route domains.